### DISCUSSION TITLE
items not vertically center aligned when it's height in flaoting number

### DISCUSSION THREAD

**Discussion by akshata456** [tailwindlabs/tailwindcss#13361] [Help]
i'm using flex classes.when design like ,
<div class="flex items-center w-4 h-4"><div class="cv-block w-[0.844rem] h-[0.844rem] bg-red"></div></div>
it not setting the child div center align why ? if we change hight value to round number then it working properly.

**Comment by alex-krasikau** [comment]
Hey @akshata456!

To center a child inside a div using flex, use `flex items-center justify-center`.

Check the complete example here: [Tailwind Play - Center Div](https://play.tailwindcss.com/1MK1zwR0up).

**Comment by akshata456** [comment]
` <div className="flex items-center w-[30px] h-[30px] bg-black justify-center">
      <div className="block w-5 h-[20.5px] bg-red-900"></div>
    </div>` tried this one but not working see top/bottom space difference

**Reply by alex-krasikau** [reply]
I tried your example in Tailwind Play, and it worked well for me: [Tailwind Play Example](https://play.tailwindcss.com/QUUgwBVEo8).

Could you share the reproduction?

**Comment by akshata456** [comment]
I testing on mac with brave browser .lmk if you're checking on same device

**Reply by alex-krasikau** [reply]
Mac, Chrome. Here's a screenshot.

<img width="220" alt="CleanShot 2024-03-26 at 18 21 16@2x" src="https://github.com/tailwindlabs/tailwindcss/assets/97882909/ca10585c-d335-4d5c-a6fc-22d04483fa1f">

The only thing that comes to mind is that the browser incorrectly displays the `20.5px` height. However, Brave is a Chromium-based browser and shouldn't have any problems with rendering subpixel elements.

**Comment by akshata456** [comment]
It's working in tailwind play , but here screenshot of demo on browser now child height is set 26.5 , it showing clear difference between top and bottom padding.
<img width="439" alt="Screenshot 2024-03-26 at 11 31 34 PM" src="https://github.com/tailwindlabs/tailwindcss/assets/34393874/ec180078-bb5f-4ffb-93cc-9ec288d39bd5">

**Reply by alex-krasikau** [reply]
Could you provide more context? It would be helpful if you could create a reproduction in Tailwind Play.

**Comment by akshata456** [comment]
@alex-krasikau https://play.tailwindcss.com/frxgpmNInb check here
<img width="425" alt="Screenshot 2024-03-27 at 7 58 46 AM" src="https://github.com/tailwindlabs/tailwindcss/assets/34393874/582064cb-f553-43f1-aa1a-7bc3e20106cf">

**Reply by alex-krasikau** [reply]
This issue is caused by using 0.5 px. If you stick to whole values, you won't encounter problems. Browsers generally support sub-pixel values, but they can be unreliable and occasionally lead to visual glitches.
